include/aspect/particle/generator/random_function.h

```diff
@@ -0,0 +1,120 @@
+/*
+ Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef __aspect__particle_generator_random_function_h
+#define __aspect__particle_generator_random_function_h
+
+#include <aspect/particle/generator/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/particle/definitions.h>
+
+#include <deal.II/base/parsed_function.h>
+
+#include <boost/random.hpp>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      /**
+       * Generate random distribution of particles over entire simulation domain.
+       * The density of particles is determined by a user-defined probability
+       * density function.
+       */
+      template <int dim>
+      class RandomFunction : public Interface<dim>, public SimulatorAccess<dim>
```

Comment:
I think `RandomFunction` is a term that's easy to misunderstand. How about `ProbabilityDensity`?